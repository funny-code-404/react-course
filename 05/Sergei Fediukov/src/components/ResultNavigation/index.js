import React from 'react'
import { Link } from 'react-router-dom'

const ResultNavigation = (props) => {
    return (
        <div className='menu_link' >
            { Boolean(props.data.length) &&
                <h1>
                    {props.title}
                </h1>
            }
            <ul >
                {Boolean(props.data.length) &&
                    props.data.map(x =>
                        <li
                            key={x.id}>
                            <Link
                                to={`${props.path}/${x.id}`}>
                                {x.name}
                            </Link>
                        </li>)
                }
            </ul>
        </div>
    )
}
export default ResultNavigation
