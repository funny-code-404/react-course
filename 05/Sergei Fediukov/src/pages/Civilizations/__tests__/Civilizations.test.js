import { configure, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import { Civilizations } from '..'
import ResultNavigation from '../../../components/ResultNavigation';

configure({ adapter: new Adapter() });

describe('Should render correctly "Civilizations"', () => {
    const mockFn = jest.fn();

    const props = {
        civilizations: { data: 'test data' },
        location: { pathname: 'test pathname' },
        history: { push: mockFn }
    }
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Civilizations {...props} />)
    })
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1)
    })
    it('component has 1 navigation button', () => {
        expect(wrapper.find('button').length).toBe(1)
    })
    it('navigation button should call onClick when clicking', () => {
        wrapper.find('button').simulate('click')
        expect(mockFn).toHaveBeenCalledTimes(1)
    })
    it('component has children div with "second_nav" class', () => {
        expect(wrapper.find('.second_nav')).toHaveLength(1)
    })
    it('component has children component "ResultNavigation"', () => {
        expect(wrapper.find(ResultNavigation)).toHaveLength(1)
    })

})