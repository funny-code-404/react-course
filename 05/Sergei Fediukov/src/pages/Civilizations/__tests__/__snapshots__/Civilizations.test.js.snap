// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly "Civilizations" to Match Snapshot 1`] = `
<div>
  <button
    data-testid="nav_button"
    onClick={[Function]}
  >
    go home
  </button>
  <div
    className="second_nav"
  >
    <ResultNavigation
      data="test data"
      path="test pathname"
      title="Civilizations"
    />
  </div>
</div>
`;
